proposals/nnnn-safe-shared-contiguous-storage.md

```diff
@@ -0,0 +1,1208 @@
+# Safe Access to Contiguous Storage
+
+* Proposal: [SE-NNNN](nnnn-safe-shared-contiguous-storage.md)
+* Authors: [Guillaume Lessard](https://github.com/glessard), [Andrew Trick](https://github.com/atrick), [Michael Ilseman](https://github.com/milseman)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Roadmap: [BufferView Roadmap](https://forums.swift.org/t/66211)
+* Bug: rdar://48132971, rdar://96837923
+* Implementation: Prototyped in https://github.com/apple/swift-collections (on branch "future")
+* Review: ([pitch 1](https://forums.swift.org/t/69888))([pitch 2](https://forums.swift.org/t/72745))
+
+## Introduction
+
+We introduce `Span<T>`, an abstraction for container-agnostic access to contiguous memory. It will expand the expressivity of performant Swift code without comprimising on the memory safety properties we rely on: temporal safety, spatial safety, definite initialization and type safety.
```

Comment:
```suggestion
We introduce `Span<T>`, an abstraction for container-agnostic access to contiguous memory. It will expand the expressivity of performant Swift code without compromising on the memory safety properties we rely on: temporal safety, spatial safety, definite initialization and type safety.
```